common/modulecontext/modulecontext.go

```diff
@@ -0,0 +1,96 @@
+package modulecontext
```

Comment:
I think there's too much mixing of concerns in this package - it is used both on the server side and the client side. That needs to be split out - the server should be very straightforward, just converting state to the proto and sending it, while the client should be most of what is in this package now, with any server-side code removed.

Reply:
Yeh agreed.
Simplified it a lot after removing modulecontext from the serverside, and removing the difference between global / non global.
There might be use still for data vs value when we have testing, but that's no longer in this PR anyway, so I've simplified that away as well.